backend/pkg/executionlayer/transformer.go

```
@@ -642,6 +658,60 @@ func transformEnsNameRegistered(chainID string, block *types.Eth1Block, res *db2
 	return nil
 }
 
+func transformConsolidationRequests(chainID string, block *types.Eth1Block, res *db2.IndexedBlock) error {
+	var requests []db2.ConsolidationRequest
+	for txIndex, tx := range block.GetTransactions() {
+		for _, log := range tx.GetLogs() {
+			if !bytes.Equal(log.Address, consolidationContractAddress.Bytes()) {
+				continue
+			}
+			// we have found a consolidation event
+			// now slice out the data
+			// source_address: Bytes20
+			// source_pubkey: Bytes48
+			// target_pubkey: Bytes48
+			requests = append(requests, db2.ConsolidationRequest{
+				SourceAddress:  log.Data[:20],
+				SourcePubKey:   log.Data[20:68],
+				TargetPubKey:   log.Data[68:116],
+				TxHash:         tx.GetHash(),
+				TxIndex:        txIndex,
+				BlockNumber:    block.GetNumber(),
+				BlockTimestamp: block.GetTime().AsTime(),
+			})
+		}
+	}
+	res.ConsolidationRequests = requests
+	return nil
```

atm this function never returns an error but if we add a validation check we can return an error from there